rdkit/Chem/FeatMaps/FeatMapUtils.py

```diff
@@ -71,32 +71,32 @@ def GetFeatFeatDistMatrix(fm, mergeMetric, mergeTol, dirMergeMode, compatFunc):
 
   """
   MergeMetric.valid(mergeMetric)
-
-  dists = [[1e8] * fm.GetNumFeatures() for _ in range(fm.GetNumFeatures())]
+  numFeatures = fm.GetNumFeatures()
+  dists = [[1e8] * numFeatures for _ in range(numFeatures)]
   if mergeMetric == MergeMetric.NoMerge:
     return dists
-  elif mergeMetric == MergeMetric.Distance:
-    mergeTol2 = mergeTol * mergeTol
-    for i in range(fm.GetNumFeatures()):
-      ptI = fm.GetFeature(i)
-      for j in range(i + 1, fm.GetNumFeatures()):
-        ptJ = fm.GetFeature(j)
-        if compatFunc(ptI, ptJ):
+  
+  # Setup distance matrix, depending on mergeMetric. 
+  benchmarkDict = { MergeMetric.Distance: mergeTol * mergeTol, MergeMetric.Overlap: mergeTol}
+  benchmark = benchmarkDict[mergeMetric]
+  def assignMatrix(matrix, i, j, value, constraint):
+    if value < constraint:
+      matrix[i][j] = value
+      matrix[j][i] = value
+
+  getFeature = fm.GetFeature
+  for i in range(numFeatures):
+    ptI = getFeature(i)
+    for j in range(i + 1, numFeatures):
+      ptJ = getFeature(j)
+      if compatFunc(ptI, ptJ):
+        if mergeMetric == MergeMetric.Distance:
           dist2 = ptI.GetDist2(ptJ)
-          if dist2 < mergeTol2:
-            dists[i][j] = dist2
-            dists[j][i] = dist2
-  elif mergeMetric == MergeMetric.Overlap:
-    for i in range(fm.GetNumFeatures()):
-      ptI = fm.GetFeature(i)
-      for j in range(i + 1, fm.GetNumFeatures()):
-        ptJ = fm.GetFeature(j)
-        if compatFunc(ptI, ptJ):
-          score = fm.GetFeatFeatScore(ptI, ptJ, typeMatch=False)
-          score *= -1 * ptJ.weight
-          if score < mergeTol:
-            dists[i][j] = score
-            dists[j][i] = score
+          assignMatrix(matrix=dists, i=i, j=j, value=dist2, constraint=benchmark)
+        elif mergeMetric == MergeMetric.Overlap:
+          score = fm.GetFeatFeatScore(ptI, ptJ, typeMatch=False) * (-1 * ptJ.weight)
+          assignMatrix(matrix=dists, i=i, j=j, value=score, constraint=benchmark)
```

Comment:
This looks correct. Was there a test that makes sure that it is?

Reply:
Whenever I want to check if a piece of code is covered in a test, I add `1/0` and see if the test break. If it does, I know the test. If not, I know that I need to be more careful.

Reply:
> Whenever I want to check if a piece of code is covered in a test, I add `1/0` and see if the test break. If it does, I know the test. If not, I know that I need to be more careful.

In that case, what do you want to make the test about? So that I can add it in.

Reply:
I'm not asking you to write a test. However, I checked and the functionality is called in unit tests. (I actually added them 5 years ago - hahaha). So we are good here.

Reply:
OMG. You are scaring me this time